core/src/main/scala-2/sttp/tapir/generic/internal/EndpointInputAnnotationsMacro.scala

```diff
@@ -23,6 +23,18 @@ class EndpointInputAnnotationsMacro(override val c: blackbox.Context) extends En
     val util = new CaseClassUtil[c.type, A](c, "request endpoint")
     validateCaseClass(util)
 
+    if (util.fields.isEmpty) {
+      val path = util.classSymbol.annotations
+        .map(_.tree)
+        .collectFirst { case Apply(Select(New(tree), _), List(arg)) if tree.tpe <:< endpointInput => arg }
+
+      if (path.isEmpty) {
+        c.abort(c.enclosingPosition, s"case class is empty and not annotated with @endpointInput")
+      } else if (path.exists(tree => tree.toString().contains('{') || tree.toString().contains('}'))) {
+        c.abort(c.enclosingPosition, s"case class is empty but has path arguments in @endpointInput ${path.get}")
+      }
+      return c.Expr[EndpointInput[A]](q"stringToPath(${path.get}).mapTo[${util.classSymbol}]")
```

Comment:
I'd prefer to avoid `return`s as over time it's tricky to understand the control flow. `if-else` are better to see what paths the code might take. Here it would probably be beneficial to extract two methods: `generateEmptyTargetEndpointInput` and `generateDefaultEndpointInput` (or sth like that, I just made up those names, maybe they aren't that great ;), and call one of these methods as the result of the `if`

Reply:
I fixed this